Core/include/Acts/Propagator/AtlasStepper.hpp

```diff
@@ -45,7 +46,9 @@ class AtlasStepper {
 
   /// @brief Nested State struct for the local caching
   struct State {
-    /// Default constructor - deleted
+    friend AtlasStepper<BField>;
+    friend ConstrainedStepControl<AtlasStepper>;
```

Comment:
Hm, the first one I get, but this second one is a bit problematic. If this already doesn't work via the stepper interface, the approach might be flawed, no?

Reply:
I have introduced (for the moment?) to not have to copy all step manipulation methods three times.

it works though if you require the Stepper to have a stepControl member that fulfills a certain concept.